rasterio/transform.py

```diff
@@ -318,29 +319,23 @@ def __init__(self):
 
     def close(self):
         self.closed = True
-    
-    def _ensure_arr_input(self, xs, ys, zs=None):
+
+    @staticmethod
+    def _ensure_arr_input(xs, ys, zs=None):
         """Ensure all input coordinates are mapped to array-like objects
         
         Raises
         ------
         TransformError
             If input coordinates are not all of the same length
         """
-        if (isinstance(xs, Iterable) and not isinstance(ys, Iterable)) or (
-            isinstance(ys, Iterable) and not isinstance(xs, Iterable)
-        ):
-            raise TransformError("Invalid inputs")
-        if not isinstance(xs, Iterable) and not isinstance(ys, Iterable):
-            xs = [xs]
-            ys = [ys]
-        if zs is None:
-            zs = [0] * len(xs)
-        elif not isinstance(zs, Iterable):
-            zs = [zs]
-        if len(set((len(xs), len(ys), len(zs)))) > 1:
-            raise TransformError("Input coordinates must be of equal length")
-        return xs, ys, zs
+        try:
+            xs, ys, zs = np.broadcast_arrays(xs, ys, 0 if zs is None else zs)
+        except ValueError:
+            raise TransformError(
+                "Input coordinates must be broadcastable to a 1d array"
+            )
+        return list(np.atleast_1d(xs)), list(np.atleast_1d(ys)), list(np.atleast_1d(zs))
```

Comment:
I'm curious if you need to cast to a `list` here or if returning a numpy array works here?

Reply:
I'm pretty sure I don't need to. The only reason I did was because the unit tests compared the results to lists. I had meant to remove the conversion but forgot.

Reply:
I've updated the pull request to remove the conversion of these arrays to lists. Now *_ensure_arr_input* returns *numpy* arrays.